android/modules/ui/src/java/ti/modules/titanium/ui/android/TiPreferencesFragment.java

```diff
@@ -0,0 +1,42 @@
+package ti.modules.titanium.ui.android;
+
+import org.appcelerator.kroll.common.Log;
+import org.appcelerator.titanium.TiApplication;
+import org.appcelerator.titanium.util.TiRHelper;
+
+import android.os.Bundle;
+import android.preference.PreferenceFragment;
+
+public class TiPreferencesFragment extends PreferenceFragment {
+    
+    private static final String TAG = "TiPreferencesFragment";
+    private static final String DEFAULT_PREFS_RNAME = "preferences";
+    
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        
+        String prefsName = DEFAULT_PREFS_RNAME;
+        if (getActivity().getIntent().hasExtra("prefsName")) {
+            String name = getActivity().getIntent().getExtras().getString("prefsName");
```

Comment:
please try to avoid duplicate method calling: getActivity().getIntent() is called twice back to back